src/pybind/mgr/balancer/module.py

```diff
@@ -519,6 +529,11 @@ def handle_command(self, inbuf, command):
                                   'current cluster')
             return (0, self.evaluate(ms, pools, verbose=verbose), '')
         elif command['prefix'] == 'balancer optimize':
+            # The GIL can be release by the active balancer, so disallow when active
+            if self.active:
+                return (-errno.EINVAL, '', 'Balancer enabled, disable to optimize manually')
```

Comment:
this breaks the cram test.
```
2019-10-24T11:07:54.103 INFO:tasks.cram.client.0.smithi052.stdout:/home/ubuntu/cephtest/archive/cram.client.0/misplaced.t: failed
2019-10-24T11:07:54.103 INFO:tasks.cram.client.0.smithi052.stdout:--- misplaced.t
2019-10-24T11:07:54.103 INFO:tasks.cram.client.0.smithi052.stdout:+++ misplaced.t.err
2019-10-24T11:07:54.103 INFO:tasks.cram.client.0.smithi052.stdout:@@ -11,16 +11,18 @@
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:   $ ceph balancer eval
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:   current cluster score [0-9]*\.?[0-9]+.* (re)
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:   $ ceph balancer optimize test_plan balancer_opt
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:+  Error EINVAL: Balancer enabled, disable to optimize manually
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:+  [22]
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:   $ ceph balancer ls
2019-10-24T11:07:54.104 INFO:tasks.cram.client.0.smithi052.stdout:-  [
2019-10-24T11:07:54.105 INFO:tasks.cram.client.0.smithi052.stdout:-      "test_plan"
2019-10-24T11:07:54.105 INFO:tasks.cram.client.0.smithi052.stdout:-  ]
2019-10-24T11:07:54.105 INFO:tasks.cram.client.0.smithi052.stdout:+  []
```

Reply:
Modified src/test/cli-integration/balancer/misplaced.t because of new limitation.  You can't use "ceph balancer optimize" while active balancer is enabled.